docs/src/user-guide/guides-retention.md

```
@@ -0,0 +1,167 @@
+:::{note}
+If the retention period is set to `null`, the corresponding garbage collection task will not run 
+even if `sweep_interval` is configured.
+:::
+
+---
+
+## Internal
+
+This section documents some of CLP’s internal behavior for retention and garbage collection.
+
+### Handling data race conditions
```

markdownlint-cli2 (0.17.2)

Headings should be surrounded by blank lines
Expected: 1; Actual: 0; Below

(MD022, blanks-around-headings)